pkg/snclient/check_logfile.go

```diff
@@ -0,0 +1,256 @@
+package snclient
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"slices"
+	"strconv"
+	"strings"
+)
+
+func init() {
+	AvailableChecks["check_logfile"] = CheckEntry{"check_logfile", NewCheckLogFile}
+}
+
+type CheckLogFile struct {
+	FilePath         []string
+	Paths            string
+	LineDelimeter    string
+	TimestampPattern string
+	ColumnDelimter   string
+	LabelPattern     []string
+}
+
+type LogLine struct {
+	LineNumber int
+	Content    string
+}
+
+func NewCheckLogFile() CheckHandler {
+	return &CheckLogFile{}
+}
+
+func (c *CheckLogFile) Build() *CheckData {
+	return &CheckData{
+		implemented:  ALL,
+		name:         "check_logfile",
+		description:  "Checks logfiles or any other text format file for errors or other general patterns",
+		detailSyntax: "%(line)", // cut to 200 chars
+		okSyntax:     "%(status) - All %(count) / %(total) Lines OK",
+		topSyntax:    "%(status) - %(problem_count)/%(count) lines (%(count)) %(problem_list)",
+		emptySyntax:  "%(status) - No files found",
+		emptyState:   CheckExitUnknown,
+		args: map[string]CheckArgument{
+			"file":         {value: &c.FilePath, description: "The file that should be checked"},
+			"files":        {value: &c.Paths, description: "Comma separated list of files"},
+			"line-split":   {value: &c.LineDelimeter, description: "Character string used to split a file into several lines (default \\n)"},
+			"column-split": {value: &c.ColumnDelimter, description: "Tab slit default: \\t"},
+			"label":        {value: &c.LabelPattern, description: "label:pattern => If the pattern is matched in a line the line will have the label set as detail"},
+		},
+		result: &CheckResult{
+			State: CheckExitOK,
+		},
+		attributes: []CheckAttribute{
+			{name: "count ", description: "Number of items matching the filter. Common option for all checks."},
+			{name: "filename ", description: "The name of the file"},
+			{name: "line", description: "Match the content of an entire line"},
+			{name: "columnN", description: "Match the content of the N-th column only if enough columns exists"},
+		},
+		exampleDefault: `
+		`,
+		exampleArgs: ``,
+	}
+}
+
+// Check implements CheckHandler.
+func (c *CheckLogFile) Check(_ context.Context, snc *Agent, check *CheckData, _ []Argument) (*CheckResult, error) {
+	c.FilePath = append(c.FilePath, strings.Split(c.Paths, ",")...)
+	if len(c.FilePath) == 0 {
+		return nil, fmt.Errorf("no file defined")
+	}
+
+	if snc.alreadyParsedLogfiles == nil {
+		snc.alreadyParsedLogfiles = make(map[string]ParsedFile, 0)
+	}
+
+	patterns := make(map[string]*regexp.Regexp, len(c.LabelPattern))
+	for _, labelPattern := range c.LabelPattern {
+		parts := strings.SplitN(labelPattern, ":", 2)
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("the label pattern is in the wrong format")
+		}
+		patterns[parts[0]] = regexp.MustCompile(parts[1])
```

Comment:
i'd say it's safer to use `regexp.Compile` here and check for an error.
The "MustCompile" is intended to use for global pkg defaults since it panics on errors.
And you don't want to panic the agent on user input.